lib/rules/no-restricted-properties.js

```diff
@@ -22,22 +22,39 @@ module.exports = {
         schema: {
             type: "array",
             items: {
-                type: "object",
-                properties: {
-                    object: {
-                        type: "string"
+                anyOf: [ // `object` and `property` are both optional, but at least one of them must be provided.
+                    {
+                        type: "object",
+                        properties: {
+                            object: {
+                                type: "string"
+                            },
+                            property: {
+                                type: "string"
+                            },
+                            message: {
+                                type: "string"
+                            }
+                        },
+                        additionalProperties: false,
+                        required: ["object"]
```

Comment:
By the way, I'm wondering if `property` should be omittable because this rule is `no-restricted-properties`